File: exercise/php/day14/tests/ChildMapperTest.php
<?php

use Children\ChildMapper;
use Children\Db2\X5T78;
use Children\DTOs\Gender;

it('map X5T78 to girl', function () {
    $db2Child = new X5T78();
    $db2Child->id = uniqid();
    $db2Child->n1 = 'Alice';
    $db2Child->n2 = 'Marie';
    $db2Child->n3 = 'Smith';
    $db2Child->cityOfBirthPc = 'Paradise';
    $db2Child->personBd = '2017-03-19';
    $db2Child->salutation = 'Girl';
    $db2Child->stNum = '123';
    $db2Child->stName = 'Sunny Street';
    $db2Child->stC = 'Paradise';
    $db2Child->stCid = '99';

    $child = ChildMapper::toDto($db2Child);

    expect($child->id)->toBe($db2Child->id)
        ->and($child->firstName)->toBe($db2Child->n1)
        ->and($child->middleName)->toBe($db2Child->n2)
        ->and($child->lastName)->toBe($db2Child->n3)
        ->and($child->birthCity)->toBe($db2Child->cityOfBirthPc)
        ->and($child->birthDate)->toBe($db2Child->personBd)
        ->and($child->gender)->toBe(Gender::Girl)
        ->and($child->address)->not->toBeNull()
        ->and($child->address->number)->toBe($db2Child->stNum)
        ->and($child->address->street)->toBe($db2Child->stName)
        ->and($child->address->city)->toBe($db2Child->stC)
        ->and($child->address->countryId)->toBe(99);
});


it('map X5T78 to child for a boy', function () {
    $db2Child = new X5T78();
    $db2Child->id = uniqid();
    $db2Child->n1 = 'Bob';
    $db2Child->n2 = '';
    $db2Child->n3 = 'Brown';
    $db2Child->cityOfBirthPc = 'Paradise';
    $db2Child->personBd = '2021-09-01';
    $db2Child->salutation = 'Boy';
    $db2Child->stNum = '9';
    $db2Child->stName = 'Oak Street';
    $db2Child->stC = 'Paradise';
    $db2Child->stCid = '98988';

    $child = ChildMapper::toDto($db2Child);

    expect($child->id)->toBe($db2Child->id)
        ->and($child->firstName)->toBe($db2Child->n1)
        ->and($child->middleName)->toBe($db2Child->n2)
        ->and($child->lastName)->toBe($db2Child->n3)
        ->and($child->birthCity)->toBe($db2Child->cityOfBirthPc)
        ->and($child->birthDate)->toBe($db2Child->personBd)
        ->and($child->gender)->toBe(Gender::Boy)
        ->and($child->address)->not->toBeNull()
        ->and($child->address->number)->toBe($db2Child->stNum)
        ->and($child->address->street)->toBe($db2Child->stName)
        ->and($child->address->city)->toBe($db2Child->stC)
        ->and($child->address->countryId)->toBe(98988);
});

File: exercise/kotlin/day14/src/test/kotlin/ChildMapperTest.kt
package children

import children.db2.X5T78
import children.dtos.Address
import children.dtos.Gender
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import java.util.*

class ChildMapperTest : StringSpec({
    val mapper = ChildMapper.INSTANCE

    "should map X5T78 to Child for a girl" {
        val db2Child = X5T78(
            id = UUID.randomUUID().toString(),
            n1 = "Alice",
            n2 = "Marie",
            n3 = "Smith",
            cityOfBirthPc = "Paradise",
            personBd = "19/03/2017",
            salutation = "Girl",
            stNum = "123",
            stName = "Sunny Street",
            stC = "Paradise",
            stCid = "99"
        )

        val child = mapper.toDto(db2Child)

        child.id shouldBe UUID.fromString(db2Child.id)
        child.firstName shouldBe db2Child.n1
        child.middleName shouldBe db2Child.n2
        child.lastName shouldBe db2Child.n3
        child.birthCity shouldBe db2Child.cityOfBirthPc
        child.birthDate shouldBe db2Child.personBd
        child.gender shouldBe Gender.Girl
        child.address shouldBe Address("123", "Sunny Street", "Paradise", 99)
    }

    "should map X5T78 to Child for a boy" {
        val db2Child = X5T78(
            id = UUID.randomUUID().toString(),
            n1 = "Bob",
            n3 = "Brown",
            cityOfBirthPc = "Paradise",
            personBd = "01/09/2021",
            salutation = "Boy",
            stNum = "9",
            stName = "Oak Street",
            stC = "Paradise",
            stCid = "98988"
        )

        val child = mapper.toDto(db2Child)

        child.id shouldBe UUID.fromString(db2Child.id)
        child.firstName shouldBe db2Child.n1
        child.middleName shouldBe db2Child.n2
        child.lastName shouldBe db2Child.n3
        child.birthCity shouldBe db2Child.cityOfBirthPc
        child.birthDate shouldBe db2Child.personBd
        child.gender shouldBe Gender.Boy
        child.address shouldBe Address("9", "Oak Street", "Paradise", 98988)
    }
})